### PR TITLE
Feature/mood meter

### PR DESCRIPTION
**Pull request by iceHtwoO**

### CODE REVIEW

**Review by TobiasAnzinger — APPROVED** [review]

**Review by waspmuc — CHANGES_REQUESTED** [review]
Great example why it is a bad idea to maintain multible branches...
You have the same changes here as in the other branch. Write small features and merge them regularly to master/develop and try to keep them atomic. You are working in a team and not on your own. Keep that in mind.

File: src/com/jambit/MoodMeter.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.jambit;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/mdm")
```
Comment (waspmuc) [inline src/com/jambit/MoodMeter.java]: there is no reason to abbreaviate that much. Readability and understandabilty is key for quality software. Try to use spelling and easy to understand names. I absolutely recommend that book for you: https://www.investigatii.md/uploads/resurse/Clean_Code.pdf

File: src/com/jambit/MoodMeter.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  @GET
+  @Path("avg")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getAvg(@DefaultValue("4") @QueryParam("t") String time) {
+    return Response.status(200).entity(time).build();
+  }
+
+  @GET
+  @Path("dist")
```
Comment (waspmuc) [inline src/com/jambit/MoodMeter.java]: use proper name

File: src/com/jambit/MoodMeter.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import javax.ws.rs.core.Response;
+
+@Path("/mdm")
+public class MoodMeter {
+
+  @GET
+  @Path("avg")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getAvg(@DefaultValue("4") @QueryParam("t") String time) {
+    return Response.status(200).entity(time).build();
```
Comment (waspmuc) [inline src/com/jambit/MoodMeter.java]: why do you return the parameter? 
You should call the service layer instead and return a proper datatype. E.g. 
moodMeterService.fetchEntriesInTime(Date time)...
The implementation of that service method could then return a not implemented exception or any empty list until you're done.

File: src/com/jambit/MoodMeter.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/mdm")
+public class MoodMeter {
+
+  @GET
+  @Path("avg")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getAvg(@DefaultValue("4") @QueryParam("t") String time) {
```
Comment (waspmuc) [inline src/com/jambit/MoodMeter.java]: you also need to handle/parse the time

**Review by waspmuc — CHANGES_REQUESTED** [review]

File: src/com/jambit/MoodMeter.java (hunk 2 above)
Comment (iceHtwoO) [reply src/com/jambit/MoodMeter.java]: is distro okay ?

File: src/com/jambit/services/MoodMeterService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    public Object run() {
+        return calculateMoodAverage(new ArrayList<MoodEntries>());
+    }
+
+
+    private float calculateMoodAverage(ArrayList<MoodEntries> moodEntries) {
+        float entryCount = moodEntries.size();
+        float entrySumm = 0;
+        for (int i = 0; i < entryCount; i++) {
+            entrySumm = entrySumm + moodEntries.get(i).vote;
```
Comment (iceHtwoO) [inline src/com/jambit/services/MoodMeterService.java]: Use x += y instead of x = x + y

File: src/com/jambit/services/MoodMeterService.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        }
+        float average = (float)(Math.round((entrySumm / entryCount) * 100)) / 100;
+        return average;
```
Comment (iceHtwoO) [inline src/com/jambit/services/MoodMeterService.java]: You can combine both lines

**Review by waspmuc — APPROVED** [review]
lass uns das mal mache ;)

**Pull request merged**